drivers/public_reports/app/models/public_reports/state_level_homelessness.rb

```diff
@@ -205,18 +211,36 @@ def parsed_pre_calculated_data
         where(client_id: report_scope.select(:client_id))
     end
 
-    private def quarter_dates
+    private def iteration_dates
       date = filter_object.start_date
       # force the start to be within the chosen date range
-      date = date.next_quarter if date.beginning_of_quarter < date
+      date = date.send(next_iteration_method) if date.send(beginning_iteration_method) < date
```

Comment:
I can be convinced otherwise, but this seems like unnecessary metaprograming? Why not just
```suggestion
      date = next_iteration(date) if beginning_iteration(date) < date
```

Reply:
Hmm, I guess that's easier to follow.  I kinda liked that it was calling it on date, but I agree with you.